src/components/modules/Projects/Projects.js

```diff
@@ -1,14 +1,35 @@
 import Link from "next/link";
+import { useState } from "react";
 import { useRouter } from "next/router";
 import Card from "../../common/Card";
 import { Layout, ProjectContainer } from "./StyledProjects";
+import SearchBar from "./Project/SearchBar";
+import Fuse from "fuse.js";
 
 const Projects = ({ projects }) => {
+  const [searchResult, setSearchResult] = useState(projects);
+
+  const options = {
+    includeScore: true,
+    keys: ["keywords.keyword"],
+    threshold: 0.3
+  };
+
+  const fuse = new Fuse(projects, options);
+
+  const searchProject = (searchQuery) => {
+    const result = fuse.search(searchQuery).map(({ item }) => item);
+    setSearchResult(result);
+  };
+
   const router = useRouter();
+
   if (router.isFallback) {
     return <div>Loading...</div>;
   }
 
+  const items = searchResult.length ? searchResult : projects
```

Comment:
if we search for something that doesn't exist then we're getting the whole array of projects again?

Reply:
I am doing some research to see how I can tackle this. It is not ideal.